Libraries/Blob/BlobTypes.js

```diff
@@ -10,7 +10,22 @@
 
 'use strict';
 
-export opaque type BlobCollector = {...};
+export opaque type BlobCollector: {...} = {...};
+
+// **Temporary workaround**
+// TODO(#24654): Use turbomodules for the Blob module.
+// Blob collector is a jsi::HostObject that is used by native to know
+// when the a Blob instance is deallocated. This allows to free the
+// underlying native resources. This is a hack to workaround the fact
+// that the current bridge infra doesn't allow to track js objects
+// deallocation. Ideally the whole Blob object should be a jsi::HostObject.
+export function createBlobCollector(blobId: string): BlobCollector {
+  if (global.__blobCollectorProvider == null) {
+    return {};
+  } else {
+    return global.__blobCollectorProvider(blobId);
+  }
+}
```

Comment:
Now when `__blobCollectorProvider` is unavailable, I need to create a placeholder blob collector object. I needed I moved this function here from `BlobManager` so flow would allow me to create instances of the `BlobCollector` opaque type.